server/game/cards/01-DTR/MaxBaine.js

```diff
@@ -0,0 +1,28 @@
+const DudeCard = require('../../dudecard.js');
+
+class MaxBaine extends DudeCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => this.controller.cardsInPlay.reduce((count, card) => {
+                if(card.hasKeyword('ranch')) {
+                    return count + 1;
+                }
+                return count;
+            }, 0) >= 2,
+            match: this,
+            effect: ability.effects.modifyControl(1)
+        });
+        this.persistentEffect({
+            targetController: 'current',

```

Comment:
State dependent effects need to have a condition, otherwise they are not reevaluated.
```suggestion
            targetController: 'current',
            condition: () => true,
```